Allow users to sort submissions (#22)

SuperCl4sh opened the pull request:
Fixes #22.

SuperCl4sh committed:
Allow users to sort submissions (#22)

SuperCl4sh commented:
Due to the nature of `select2.js`, users are able to enter multiple sort options, which is entirely useless since the probability of having a tie to break is near zero. In the future, perhaps it would be worth resolving this minor issue.

zecookiez commented:
Looks promising, but the sort option seems to be broken on the Best Submissions tab:

![image](https://user-images.githubusercontent.com/13069724/184503707-213c3c3b-a24b-4e1e-911b-a47fd6dc15d0.png)

Sorting seems to be going in descending order on the All submissions tab, not sure if that's intended?

SuperCl4sh committed:
Fix sorting to override sorting of submissions in 'Best Solutions' page

SuperCl4sh commented:
I believe it appeared to be sorted that way due to the Best Solution tab overriding the sorting options with runtime ascending. I have changed this and added ascending and descending sorting options for memory and runtime. In the future, perhaps a sorting option for submission source size could be added.
